### PR TITLE
fix errors in vscode pyright linter

### PR DESCRIPTION
**Pull request by PCSwingle**
@granawkins I added a new CTAG type; you were previously checking for if line number or name was None; in my own testing I couldn't find anywhere where it would've been None, but I don't have much experience with ctags. Do you know if it can ever be None?

### CODE REVIEW

**Comment by granawkins** [comment]
From the [documentation](https://docs.ctags.io/en/latest/man/ctags.1.html#tags-file-contents-options) and a couple tests, I think you're correct, line number shouldn't ever be None.

File: mentat/code_map.py [hunk 1]
```diff
@@ -6,10 +6,10 @@
 from pathlib import Path
 from typing import Any
 
+CTAG = tuple[str | None, str, str, str | None, int]
```
Comment (granawkins) [inline mentat/code_map.py:9]: I like this having its own class. Does it make sense to label these fields, just for clarity - maybe with attr or namedtuple or something:

File: mentat/code_map.py [hunk 2]
```diff
@@ -6,10 +6,10 @@
...
 
-def get_ctags(
-    abs_file_path: Path, exclude_signatures: bool = False
-) -> set[tuple[str | int | None, ...]]:
+
+def get_ctags(abs_file_path: Path, exclude_signatures: bool = False):
```
Comment (granawkins) [inline mentat/code_map.py:12]:
```suggestion
def get_ctags(abs_file_path: Path, exclude_signatures: bool = False) -> set[CTAG]:
```

**Review by granawkins — APPROVED** [review]
Couple small suggestions, otherwise looks good.

File: mentat/code_map.py (hunk 1 above)
Comment (PCSwingle) [reply mentat/code_map.py:9]: I think it makes more sense as a type; it's not like we're doing anything with it other than getting the internals

**Pull request merged**